spec/std/system/group_spec.cr

```diff
@@ -0,0 +1,89 @@
+require "spec"
+require "system/group"
+
+GROUP_NAME = {{ `id -gn`.stringify.chomp }}
+GROUP_ID   = {{ `id -g`.stringify.chomp }}
+
+describe System::Group do
+  describe "find_by" do
```

Comment:
Just a minor detail: `find_by(*, name)` and `find_by(*, id)` are entirely different methods. I wouldn't group them in the same describe block. Instead, the nilable and not-nilable variants can be grouped because they do the exact same thing and only differ in error handling.